src/Documenter.jl

```diff
@@ -176,26 +176,78 @@ ignored.
 
-Documenter is designed to support multiple output formats. By default it is creates a set of
-HTML files, but the output format can be controlled with the `format` keyword. The different
-output formats may require additional keywords to be specified via plugins. The keywords for
-the default HTML output are documented for the [`Writers.HTMLWriter.HTML`](@ref) type.
-
-The default `:html` output format creates a set of HTML files, but Documenter is designed to
-support multiple output formats. Via plugin packages, Documenter also supports e.g. Markdown
-/ MkDocs and LaTeX / PDF outputs. See the [Other Output Formats](@ref) for more information.
+There are other possible formats that are enabled by using other addon-packages.
+For examples, the `DocumenterMarkdown` package define the `DocumenterMarkdown.Markdown()`
+format for use with e.g. MkDocs, and the `DocumenterLaTeX` package define the
+`DocumenterLaTeX.LaTeX()` format for LaTeX / PDF output.
+See the [Other Output Formats](@ref) for more information.
 
 # See Also
 
 A guide detailing how to document a package using Documenter's [`makedocs`](@ref) is provided
 in the [setup guide in the manual](@ref Package-Guide).
 """
-function makedocs(components...; debug = false, kwargs...)
-    document = Documents.Document(components; kwargs...)
+function makedocs(components...; debug = false, format = HTML(),
+                  html_prettyurls::Union{Bool, Nothing} = nothing, # deprecated
+                  html_disable_git::Union{Bool, Nothing} = nothing, # deprecated
+                  html_edit_branch::Union{String, Nothing} = nothing, # deprecated
+                  html_canonical::Union{String, Nothing} = nothing, # deprecated
+                  kwargs...)
+    # html_ keywords deprecation
+    html_keywords = Dict()
+    @noinline function html_warn(kw)
+        Base.depwarn("""
+        The `$kw` keyword argument should now be specified in the
+        `Documenter.HTML()` format specifier. To fix this warning replace
+        ```
+        $kw = ...
+        ```
+        with
+        ```
+        format = Documenter.HTML($kw = ...)
+        ```
+        """, :makedocs)
+    end
+    if html_prettyurls !== nothing
+        html_warn("html_prettyurls")
+        html_keywords[:prettyurls] = html_prettyurls
+    end
+    if html_disable_git !== nothing
+        html_warn("html_disable_git")
+        html_keywords[:disable_git] = html_disable_git
+    end
+    if html_edit_branch !== nothing
+        html_warn("html_edit_branch")
+        html_keywords[:edit_branch] = html_edit_branch
+    end
+    if html_canonical !== nothing
+        html_warn("html_canonical")
+        html_keywords[:canonical] = html_canonical
+    end
+    # deprecation of format as Symbols
+    function fmt(f)
+        if format === :html
+            Base.depwarn("`format = :html` is deprecated, use `format = Documenter.HTML()` instead.", :makedocs)
+            return Writers.HTMLWriter.HTML(html_keywords...)
```

Comment:
Right now, if you set `makedocs(html_something = foo, format = Documenter.HTML())`, you would get the warning, but `html_something` would not have any effect, right? Most importantly, it affects the case where you are relying on the default value for `format`.

I think we should do the following:

* Have the default for `format` be `:html` (but keep docs as they are, this will just be an implementation detail). So, in case the user passes any of the `html_*` arguments without explicitly specifying `format`, we'll set the `Documenter.HTML` object fields appropriately.
* In the `html_*` checks above, if `format != :html` or maybe if `format isa Documenter.HTML`, print out a big warning saying that the arguments do not have any effect.

Reply:
Yea, you will still get the warnings, and if you have specified the `html_` ones they are probably the values you want.